src/ln/msgs.rs

```diff
@@ -372,6 +371,12 @@ pub enum NetAddress {
 		/// The port on which the node is listenting
 		port: u16,
 	},
+	/// An unsupported address.
+	/// At deserialization, we retrieve first byte of it if needed by caller.
+	Unknown {
```

Comment:
Hmmm, the panics introduced in get_id/len are kinda annoying. A user may deserialize a NetAddress and then try to reserialize it and get a panic. It may be easier to get a higher-level enum that selects between an unknown byte and a NetAddress for the deserialization.

Reply:
But an enum doesn't fit into Readable trait signature so seems to me here we are constrained between extending NetAddress enum type with UnknownAddress  (but gross because of panic) or adding an UnknownAddressDescriptor with byte field in DecodeError ?

Reply:
Hmm? We can implement Readable for NetAddressReadEnum instead of for NetAddress and just map it properly in the message thats calling it.